assessor/lib/helmrepo.rb

```diff
@@ -0,0 +1,107 @@
+
+def helm_repo_shutdown
+  # After a test we shut the local helm repository down again. We ignore failures.
+  set errexit: false do
+    run "cf", "delete",       "-f", helm_app
+    run "cf", "delete-space", "-f", "mb-charting"
+    run "cf", "delete-org",   "-f", "mb-charting"
+  end
+  FileUtils.remove_dir(appdir, force = true)
+end
+
+def helm_repo_setup(chart, engine, chartlocation)
+  # Assemble and run node-env app serving the helm repository.
+  # I. Copy original app into fresh directory
+  FileUtils.remove_dir(appdir, force = true)
+  FileUtils.cp_r(appsrc, appdir)
+
+  # II. Change app name to something more suitable
+  m = YAML.load_file (manifest)
+  m['applications'][0]['name'] = helm_app
+  File.write(manifest, m.to_yaml)
+
+  # III. Write remote helm chart to local file
+  get_engine_chart(chartlocation)
+
+  # IV. Patch local chart archive. Stop on failure
+  return "" unless sucessfully_patched_chart(engine)
+
+  # V. Place index (*) and patched chart.
+  #    (*) With proper chart archive reference
+  File.write(chart_index, make_index_yaml(chart, engine, chart_ref))
+  FileUtils.cp(archive_patched, archive_app)
+
+  # VI. Start repository (push app)
+
+  run "cf", "api", "--skip-ssl-validation", target
+  run "cf", "auth", "admin", @auth
+  run "cf create-org   mb-charting"
+  run "cf target    -o mb-charting"
+  run "cf create-space mb-charting"
+  run "cf target    -o mb-charting"
+  run "cf enable-feature-flag diego_docker"
+
+  FileUtils.cd(appdir) do
+    run "cf", "push", "-n", helm_app
```

Comment:
Use `cf push -p #{appdir} -n …` instead of `cd`?

Reply:
will try. this one will need actual testing

Reply:
Done with commit 2361a7aa0b0354bf8b47cec5b4fdaf0048ab588c.
Actual command is `cf push appname -p appdir -n appname` where the `-n ...` is the hostname for the route.